filebeat/input/filestream/internal/task/group.go

```diff
@@ -0,0 +1,129 @@
+package task
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type Group struct {
+	limit uint64
+
+	sem *semaphore.Weighted
+	wg  *sync.WaitGroup
+
+	errs     []error
+	errsSize uint64
+	errsNext uint64
+	errsMu   *sync.Mutex
+
+	ctx       context.Context
+	cancelCtx context.CancelFunc
+}
+
+type Goer interface {
+	Go(fn func(context.Context) error) error
```

Comment:
Please add godoc for this method.  I know it exists on the implementation below but I think some of the semantics (e.g. should not block) would be good to be documented as part of the interface.

Reply:
I see you removed the `Goer` interface altogether.  So this thread is no longer relevant.